src/main/scala/com/vngrs/etl/Transform.scala

```diff
@@ -0,0 +1,64 @@
+package com.vngrs.etl
+
+import org.apache.spark.rdd.RDD
+
+import scala.reflect.ClassTag
+
+/**
+  * Transforms data ([[org.apache.spark.rdd.RDD]]) to another form.
+  *
+  * @tparam A Type of the data ([[org.apache.spark.rdd.RDD]]).
+  */
+trait Transform[A, B] {
+
+  /**
+    * Runs transformation operation,
+    *
+    * @param input [[org.apache.spark.rdd.RDD]]
+    * @return Transformed [[org.apache.spark.rdd.RDD]]
+    */
+  def apply(input: RDD[A]): RDD[B]
+}
+
+/**
+  * Companion object which acts as a Factory.
+  */
+object Transform {
```

Comment:
The same technique could be applied here too. Just create a more general apply method with signature `RDD[A] => RDD[B]` and implement map and filter with that